crates/sui-adapter/src/execution_engine.rs

```diff
@@ -186,7 +185,7 @@ fn execute_transaction<
     let mut result = result.and_then(|()| {
         let mut execution_result = execution_loop::<Mode, _>(
             temporary_store,
-            transaction_kind,
+            transaction_kind.clone(),
```

Comment:
I would be quite surprised if compiler cannot detect that this clone is unnecessary, but fair :)

Reply:
True--we need to read a couple of fields, but we could also do that in advance.

I haven't spent a lot of time looking at Rust's compiled output recently, but have definitely been bitten by big clones in the past and try to avoid them.

Reply:
yeah I looked as well when I saw the clone but we need the info, I would almost say one option would be to clone only when needed instead of always (so specifically only for certain system transaction or change epoch).
I do not believe the compiler would be able to optimize that clone to be honest, though through a load of optimizations piling up maybe if we are lucky?